plugins/auth-backend-module-pinniped-provider/src/authenticator.test.ts

```diff
@@ -276,10 +299,133 @@ describe('pinnipedAuthenticator', () => {
               url: 'test',
             },
           } as unknown as OAuthAuthenticatorStartInput,
-          implementation,
+          authCtx,
         ),
       ).rejects.toThrow('authentication requires session support');
     });
+
+    it('refreshes oidc metadata after a failed fetch', async () => {
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, _ctx) => res.networkError('Timeout'),
+        ),
+      );
+
+      const authCtxCreatedWhileSupervisorUnavailable =
+        pinnipedAuthenticator.initialize({
+          callbackUrl: 'https://backstage.test/callback',
+          config: new ConfigReader({
+            federationDomain: 'https://federationDomain.test',
+            clientId: 'clientId',
+            clientSecret: 'clientSecret',
+          }),
+        });
+
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, ctx) =>
+            res(
+              ctx.status(200),
+              ctx.set('Content-Type', 'application/json'),
+              ctx.json(issuerMetadata),
+            ),
+        ),
+      );
+
+      const response = await pinnipedAuthenticator.start(
+        startRequest,
+        authCtxCreatedWhileSupervisorUnavailable,
+      );
+
+      expect(response.url).toMatch('https://pinniped.test/oauth2/authorize');
+    });
+
+    it('caches oidc metadata after a success', async () => {
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, _ctx) => res.networkError('Timeout'),
+        ),
+      );
+
+      const authCtxCreatedWhileSupervisorUnavailable =
+        pinnipedAuthenticator.initialize({
+          callbackUrl: 'https://backstage.test/callback',
+          config: new ConfigReader({
+            federationDomain: 'https://federationDomain.test',
+            clientId: 'clientId',
+            clientSecret: 'clientSecret',
+          }),
+        });
+
```

Comment:
Since this test is about what happens after a success, I don't think we _also_ need to start with a failure. To my mind starting out with a failure state makes this a little harder to follow
```suggestion
```

Reply:
ok i can see how that would be confusing, I will make the change

Reply:
Refactored and changed.